fakestorage/object.go

```diff
@@ -934,17 +934,31 @@ func (s *Server) patchObject(r *http.Request) jsonResponse {
 	vars := unescapeMuxVars(mux.Vars(r))
 	bucketName := vars["bucketName"]
 	objectName := vars["objectName"]
-	var metadata struct {
+
+	type acls struct {
+		Entity string
+		Role   string
+	}
+
+	var dataInBody struct {
 		Metadata map[string]string `json:"metadata"`
+		Acl      []acls
 	}
-	err := json.NewDecoder(r.Body).Decode(&metadata)
+	err := json.NewDecoder(r.Body).Decode(&dataInBody)
 	if err != nil {
 		return jsonResponse{
 			status:       http.StatusBadRequest,
 			errorMessage: "Metadata in the request couldn't decode",
 		}
 	}
-	backendObj, err := s.backend.PatchObject(bucketName, objectName, metadata.Metadata)
+	backendObj, err := s.backend.PatchObject(bucketName, objectName, dataInBody.Metadata)
+	if len(dataInBody.Acl) > 0 {
+		backendObj.ACL = []storage.ACLRule{}
+		for _, aclData := range dataInBody.Acl {
+			newAcl := storage.ACLRule{Entity: storage.ACLEntity(aclData.Entity), Role: storage.ACLRole(aclData.Role)}
+			backendObj.ACL = append(backendObj.ACL, newAcl)
+		}
```

Comment:
Hmm this won't work with the backend filesystem, will it? We're modifying the value in memory and not persisting it. We need to update the PatchObject signature to take something like "attrsToUpdate" and apply changes to other attributes, not just metadata (this is necessary for #1024 too).

Reply:
Ah good point. I will fix this and push the changes.

Reply:
I have made the requested change. Let me know if what I did was best practice or if there is a better way.